incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -184,25 +182,17 @@ func (r *HierarchyConfigReconciler) onMissingNamespace(log logr.Logger, ns *fore
 		return false
 	}
 
+	// If the namespace is removed, no more action need on the reconciliation. The
+	// namespace is only unset existence below.
 	if !ns.Exists() {
-		// The namespace doesn't exist on the server, but its owner expects it to be there. Initialize
-		// it so it gets created; once it is, it will be reconciled again.
-		if ns.Owner != "" {
-			log.Info("Will create missing namespace", "forOwner", ns.Owner)
-			nsInst.Name = ns.Name()
-			// Set "api.AnnotationOwner" annotation to the non-existent yet namespace.
-			metadata.SetAnnotation(nsInst, api.AnnotationOwner, ns.Owner)
-		}
 		return true
 	}
 
 	// Remove it from the forest and notify its relatives
 	r.enqueueAffected(log, "relative of deleted namespace", ns.RelativesNames()...)
 	// Enqueue the HNS if the owned namespace is deleted.
-	if r.HNSReconcilerEnabled {
-		if ns.Owner != "" {
-			r.hnsr.enqueue(log, ns.Name(), ns.Owner, "hns for the deleted owned namespace")
-		}
+	if ns.Owner != "" {
+		r.hnsr.enqueue(log, ns.Name(), ns.Owner, "hns for the deleted owned namespace")
```

Comment:
An alternative option: have the HNSR put a watch on namespaces as well? Then K8s would invoke HNSR directly and we wouldn't have to do it here?

Reply:
Yes, that sg. We can have a mapping func to trigger HNSR on related HNSes.